pinot-core/src/test/java/com/linkedin/pinot/queries/FastHllQueriesTest.java

```diff
@@ -0,0 +1,257 @@
+    // Test query with filter
+    aggregationOperator = getOperatorForQueryWithFilter(BASE_QUERY);
+    resultsBlock = (IntermediateResultsBlock) aggregationOperator.nextBlock();
+    executionStatistics = aggregationOperator.getExecutionStatistics();
+    Assert.assertEquals(executionStatistics.getNumDocsScanned(), 6129L);
+    Assert.assertEquals(executionStatistics.getNumEntriesScannedInFilter(), 85248L);
+    Assert.assertEquals(executionStatistics.getNumEntriesScannedPostFilter(), 12258L);
+    Assert.assertEquals(executionStatistics.getNumTotalRawDocs(), 30000L);
+    aggregationResult = resultsBlock.getAggregationResult();
+    Assert.assertEquals(((HyperLogLog) aggregationResult.get(0)).cardinality(), 17L);
+    Assert.assertEquals(((HyperLogLog) aggregationResult.get(1)).cardinality(), 1197L);
+    // Test query with group-by
+    AggregationGroupByOperator aggregationGroupByOperator = getOperatorForQuery(BASE_QUERY + GROUP_BY);
+    resultsBlock = (IntermediateResultsBlock) aggregationGroupByOperator.nextBlock();
+    executionStatistics = aggregationGroupByOperator.getExecutionStatistics();
+    Assert.assertEquals(executionStatistics.getNumDocsScanned(), 4613L);
+    Assert.assertEquals(executionStatistics.getNumEntriesScannedInFilter(), 0L);
+    Assert.assertEquals(executionStatistics.getNumEntriesScannedPostFilter(), 13839L);
+    Assert.assertEquals(executionStatistics.getNumTotalRawDocs(), 30000L);
+    AggregationGroupByResult aggregationGroupByResult = resultsBlock.getAggregationGroupByResult();
+    GroupKeyGenerator.GroupKey firstGroupKey = aggregationGroupByResult.getGroupKeyIterator().next();
+    Assert.assertEquals(firstGroupKey.getStringKey(), "");
+    Assert.assertEquals(((HyperLogLog) aggregationGroupByResult.getResultForKey(firstGroupKey, 0)).cardinality(), 21L);
+    Assert.assertEquals(((HyperLogLog) aggregationGroupByResult.getResultForKey(firstGroupKey, 1)).cardinality(), 691L);
+
+    // Test inter segments base query
+    BrokerResponseNative brokerResponse = getBrokerResponseForQuery(BASE_QUERY);
+    QueriesTestUtils.verifyAggregationResult(brokerResponse, 4L, 0L, 8L, 120000L, new String[]{"21", "1762"});
+    // Test inter segments query with filter
+    brokerResponse = getBrokerResponseForQueryWithFilter(BASE_QUERY);
+    QueriesTestUtils.verifyAggregationResult(brokerResponse, 24516L, 340992L, 49032L, 120000L,
+        new String[]{"17", "1197"});
+    // Test inter segments query with group-by
+    brokerResponse = getBrokerResponseForQuery(BASE_QUERY + GROUP_BY);
+    QueriesTestUtils.verifyAggregationResult(brokerResponse, 18452L, 0L, 55356L, 120000L, new String[]{"21", "1762"});
+
+    deleteSegment();
+  }
+
+  @Test
+  public void testFastHllWithLog2m8() throws Exception {
+    buildSegment(8);
+
+    // Test inner segment queries
+    // Test query without filter
+    AggregationOperator aggregationOperator = getOperatorForQuery(BASE_QUERY);
```

Comment:
[Suggestion] It's better to use TestNG's DataProvider to provide the segment(s), query and result validator for the test. That way, we can define additional entry in data provider if our segment format or set of queries changes. All the tests can be run for those new formats by defining data provider. We won't have to change actual tests.

This was a major handicap when we added v3 format - there was no easy way to verify that all tests work with v3 format.